### PR TITLE
New lured pokemon support

### PR DESCRIPTION
**Pull request by onilton**
## Description
I updated and rebased lured pokemon support with the current develop (lot of breaking changes) 
Based on the work from @maddhatter and @wobblybobz , I used #1417 as the base for this.

Add lured Pokemon to the map adapted from maddhatter PR1079.
Adding support for Encounters introduced in PR1272

The lured Pokemon is added to the pokemon table with a null spawnpoint_id. map.js marks Pokemon with no spawnpoint_id as lured on the info popup for the Pokemon. Other than the addition of "Lured Pokemon" to their popup, there is no front-end distinction between lured and normal Pokemon.

## Motivation and Context
Adding support for Lured Pokemon again

## How Has This Been Tested?
Tested in my own private map

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by hokiepokedad** [comment]
Would this PR require a rescan of spawnpoints?

**Comment by lluisd** [comment]
I've been using that PR during last month to monitor my neighbour's lured pokestop. It works good but I think that it has some things to improve:

- Put some parameter like "lure: true" and "lure-whitelist: [pokestop_id, pokestop_id, pokestop_id]". Also add the pokestop_id in the website pokestop marker.
- When a lured pokestop is detected, scan only each 3 minutes (when the pokemon appears, you can calculate it) and do some "jitter" arround the pokestop area. This will avoid captchas.

What do you think?

**Comment by onilton** [comment]
@hokiepokedad No

**Comment by onilton** [comment]
@lluisd I liked your idea of whitelist and showing pokestop_id in the map. Will try it in the near future

About the second point, you would require some work in scheduler and would complicate things a little bit. 

My idea for now is for people to just setup an RM instance checking those spots every x seconds.

Usually far pokestops are not interesting since you can't get there quick enough, you usually want to monitor some pokestops around you.

**Comment by tobias86aa** [comment]
Should be nice if something like this could be merge :)

**Comment by byjurbraam** [comment]
Hey! We would really appreciate to have this feature. A lot of people are asking for lured pokemon support. Can we help you?

**Comment by onilton** [comment]
Added flag to enabled/disable lured pokemon. Made by @Arengo .

File: static/js/map.common.js [hunk 1]
```diff
@@ -1041,15 +1041,17 @@ function setupPokemonMarker(item, map, isBounceDisabled) {
...
     if (isBounceDisabled === true) {
         animationDisabled = true
     }
 
     var marker = new google.maps.Marker({
         position: {
-            lat: item['latitude'],
-            lng: item['longitude']
+            lat: (!isLured) ? item['latitude'] : (item['latitude'] + 0.0001),
+            lng: (!isLured) ? item['longitude'] : (item['longitude'] + 0.0001)
```
Comment (Arengo) [inline static/js/map.common.js]: Is the marker re-position needed because otherwise the map won't update the marker?

**Comment by byjurbraam** [comment]
Hey, what needs to be done to get this merged into RocketMap? I can help if you like. This feature is nice!

**Comment by sebastienvercammen** [comment]
Closed, this is a very old PR. If this PR is still related to current code and the author rebases it, it can be re-opened on request.

**Pull request closed without merging**